hw/ip/otbn/data/bignum-insns.yml

```diff
@@ -774,13 +774,15 @@
   syntax: |
     <grd>[<grd_inc>], <offset>(<grs1>[<grs1_inc>])
   doc: |
-    Calculates a byte memory address by adding the offset to the value in the GPR `grs1`.
-    The value from this memory address is then copied into the WDR pointed to by the value in GPR `grd`.
+    Load a WLEN-byte little-endian value from memory.
+
+    The load address is `offset` plus the value in the GPR `grs1`.
+    The loaded value is stored into the WDR given by the bottom 5 bits of the GPR `grd`.
 
     After the operation, either the value in the GPR `grs1`, or the value in `grd` can be optionally incremented.
 
-    - If `grs1_inc` is set, the value in `grs1` is incremented by the value WLEN/8 (one word).
-    - If `grd_inc` is set, the value in `grd` is incremented by the value 1.
+    - If `grs1_inc` is set, the value in `grs1` is incremented by value WLEN/8 (one word).
```

Comment:
From the discussion in https://github.com/lowRISC/opentitan/issues/3587 I had intended that the `grs1` increment would have the same masking behaviour as `grd` so we want something like

> If `grs1_inc` is set, the value in `grs1` is updated to be `(*grs1 + WLEN/8) & (DMEM_SIZE - 1)`

Though that wording's slightly awkward (especially as it's broken unless `DMEM_SIZE` is a power of 2)

If we don't want to do this that's fine though does mean we need a larger incrementer so we can increment across the full 32-bit width. In which case we should probably not have the mask for the `grs1_inc` either and increment across the full 32-bit width. I don't really mind which option we go for but think we should be consistent (either both mask or neither mask).

Reply:
I definitely think we start by doing the thing that takes the smallest implementation (so what you've described). The only remaining question is how to spec it most cleanly. Thanks for reminding me - I'll try again now! :-)